### ISSUE TITLE
test_fs_traverse  fails while building/testing on conda-forge

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#3491]
The same build as in #3490 (0.11.5, conda-forge) so might all relate
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=44838

```
======================================================================
ERROR: datalad.interface.tests.test_ls_webui.test_fs_traverse
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/tests/utils.py", line 442, in newfunc
    return t(*(arg + (d,)), **kw)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/interface/tests/test_ls_webui.py", line 70, in test_fs_traverse
    annex.add(opj(topdir, 'dir'))
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/gitrepo.py", line 320, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 1486, in add
    git_options=git_options, annex_options=annex_options, update=update
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 1561, in add_
    for rec in to_be_added_recs
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 1559, in <dictcomp>
    expected_additions = {
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 1550, in _get_to_be_added_recs
    raise e
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 1531, in _get_to_be_added_recs
    files, git_options=_git_options, update=update):
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/gitrepo.py", line 1144, in add_
    to_options(update=update) + ['--verbose']
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 1269, in _git_custom_command
    return super(AnnexRepo, self)._git_custom_command(*args, **kwargs)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/gitrepo.py", line 320, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/gitrepo.py", line 1803, in _git_custom_command
    expect_fail=expect_fail)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/support/gitrepo.py", line 1840, in _run_command_files_split
    *args, **kwargs)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/cmd.py", line 693, in run
    cmd, env=self.get_git_environ_adjusted(env), *args, **kwargs)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1561498363479/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.7/site-packages/datalad/cmd.py", line 552, in run
    raise CommandError(text_type(cmd), msg, status, out[0], out[1])
datalad.support.exceptions.CommandError: CommandError: command '['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'add', '--dry-run', '-N', '--ignore-missing', '--verbose', '--', 'dir']' failed with exitcode 128
Failed to run ['git', '-c', 'receive.autogc=0', '-c', 'gc.auto=0', 'add', '--dry-run', '-N', '--ignore-missing', '--verbose', '--', 'dir'] under '/tmp/datalad_temp_tree_test_fs_traversehn8qczp_'. Exit code=128. out=add 'dir/.fgit/ab.txt'
add 'dir/subdir/file1.txt'
add 'dir/subdir/file2.txt'
 err=error: 'dir/subgit/' does not have a commit checked out
fatal: adding files failed
```

**Comment by kyleam** [comment]
This is addressed by 2127bdb01 (TST: ls_webui: Don't try to add sub-repo without commits, 2019-06-25) in gh-3476.  See that PR for a fuller description of the changes in Git.  However, that commit should be on the 0.11.x branch, so I'll open a PR with that commit alone.

**Comment by yarikoptic** [comment]
oh, great, thanks @kyleam !
